CONTRIBUTING.md

```diff
@@ -23,12 +29,129 @@ tox run -e integration   # integration tests
 tox                      # runs 'format', 'lint', 'static', and 'unit' environments
 ```
 
-## Build the charm
+### Committing
 
-Build the charm in this git repository using:
+This repo uses CI/CD workflows as outlined by
+[operator-workflows](https://github.com/canonical/operator-workflows). The four
+workflows are as follows:
 
-```shell
-charmcraft pack
+- `test.yaml`: This is a series of tests including linting, unit tests and
+  library checks which run on every pull request.
+- `integration_test.yaml`: This runs the suite of integration tests included
+  with the charm and runs on every pull request.
+- `test_and_publish_charm.yaml`: This runs either by manual dispatch or on every
+  push to the main branch or a special track/\*\* branch. Once a PR is merged
+  with one of these branches, this workflow runs to ensure the tests have passed
+  before building the charm and publishing the new version to the edge channel
+  on Charmhub.
+- `promote_charm.yaml`: This is a manually triggered workflow which publishes
+  the charm currently on the edge channel to the stable channel on Charmhub.
+
+These tests validate extensive linting and formatting rules. Before creating a
+PR, please run `tox` to ensure proper formatting and linting is performed.
+
+### Deploy
+
+This charm is used to deploy Airbyte server in a k8s cluster. For a local
+deployment, follow the following steps:
+
+#### Install Microk8s
+```bash
+# Install Microk8s from snap:
+sudo snap install microk8s --classic --channel=1.25
+
+# Install charmcraft from snap:
+sudo snap install charmcraft --classic
+
+# Add the 'ubuntu' user to the Microk8s group:
+sudo usermod -a -G microk8s ubuntu
+
+# Give the 'ubuntu' user permissions to read the ~/.kube directory:
+sudo chown -f -R ubuntu ~/.kube
+
+# Create the 'microk8s' group:
+newgrp microk8s
+
+# Enable the necessary Microk8s addons:
+microk8s enable hostpath-storage dns
 ```
 
-<!-- You may want to include any contribution/style guidelines in this document>
+#### Set up the Juju OLM
+```bash
+# Install the Juju CLI client, juju:
+sudo snap install juju --classic
+
+# Install a "juju" controller into your "microk8s" cloud:
+juju bootstrap microk8s airbyte-controller
+
+# Create a 'model' on this controller:
+juju add-model airbyte
+
+# Enable DEBUG logging:
+juju model-config logging-config="<root>=INFO;unit=DEBUG"
+
+# Check progress:
+juju status --relations --watch 2s
+juju debug-log
+```
+
+#### Deploy Charm
+```bash
+# Pack the charm:
+charmcraft pack [--destructive-mode]
```

Comment:
Might be worth explaining or linking to an explanation of when someone might require the `--destructive-mode` flag